src/store/actions.js

```diff
@@ -38,29 +38,32 @@ export default {
     if (ids.length) {
       return fetchItems(ids)
         .then(items => {
-          if (items.every(item => item.type === 'story')) {
+          // Don't fetch similar if all items are Jobs (on /jobs)
+          // In other tabs there is rarely many jobs
+          let stories = items.filter(x => x.type === 'story')
+          let otherItems = items.filter(x => x.type !== 'story')
 
-            return fetchSimilar(items.map(item => item.title))
-              .then(similar => items.map((item, idx) => {
-                item.similar = similar[idx];
-                return item;
-              }))
-              // Start fetching similar posts (potential performance issue...)
-              .then((items) => {
-                return fetchItems(items.map(i => i.similar).flat().map(sim => sim.id))
-                  .then(similarItems => {
-                    items.forEach(item => {
-                      item.similar = item.similar.map(sim => {
-                        const simItem = similarItems.find(si => si.id === sim.id);
-                        return Object.assign({ similarity_score: sim.score }, simItem);
-                      });
+          return fetchSimilar(stories.map(item => item.title))
+            .then(similar => items.map((item, idx) => {
+              item.similar = similar[idx];
+              return item;
+            }))
+            // Start fetching similar posts (potential performance issue...)
+            .then((stories) => {
+              return fetchItems(stories.map(i => i.similar).flat().map(sim => sim.id))
+                .then(similarItems => {
+                  stories.forEach(item => {
+                    item.similar = item.similar.map(sim => {
+                      const simItem = similarItems.find(si => si.id === sim.id);
+                      return Object.assign({ similarity_score: sim.score }, simItem);
                     });
-                    return items;
                   });
-              });
-              // Stop fetching similar posts (potential performance issue...)
-          }
-          return items;
+                  return stories;
+                });
+            })
+            .then((stories) => { return stories.concat(otherItems) });
```

Comment:
by concating the items like this your reordering the results as well, I'm not sure if there is any sorting in the FE to mitigare this potential issue.

I  think you migth be fine to just return the `items` variable since it's just an array of references to the objects you're modifining anyway.

Reply:
good point - I thought it was later referring to the items by the id - but what you are saying seems to make sense.

Reply:
let me try if works